src/main/resources/db/migration/V021__AddStateToCaseTable.sql

```diff
@@ -0,0 +1,24 @@
+CREATE TYPE public.CASE_STATE AS ENUM (
+    'OPEN',
+    'PENDING_CLOSURE',
+    'CLOSED'
+);
+
+ALTER TABLE public.cases
+    ADD COLUMN state CASE_STATE,
+    ADD COLUMN closed_at TIMESTAMPTZ;
+
+-- Update existing rows with a default state value
+UPDATE public.cases
+  SET state = 'OPEN'
+  WHERE state IS NULL;;
+
+-- Ensure 'state' column has the default value and is not null
+ALTER TABLE public.cases
+    ALTER COLUMN state SET DEFAULT 'OPEN',
+    ALTER COLUMN state SET NOT NULL;
```

Comment:
This can be simplified
```suggestion
ALTER TABLE public.cases
ADD COLUMN state CASE_STATE NOT NULL DEFAULT 'OPEN',
ADD COLUMN closed_at TIMESTAMPTZ;
```